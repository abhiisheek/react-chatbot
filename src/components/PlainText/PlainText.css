@import '../_colors.css';
@import '../_variables.css';

.wrapper {
  font-size: var(--font-size-tiny);
}