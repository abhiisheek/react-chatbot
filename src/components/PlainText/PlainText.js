import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import styles from './PlainText.css';

/** Component for displaying plain text response */
class PlainText extends PureComponent {
  render() {
    return (
      <div className={classNames('textMsg', styles.wrapper)}>
        {this.props.data}
      </div>
    );
  }
}

PlainText.displayName = 'PlainText';

PlainText.propTypes = {
  /** Plain text response to be displayed */
  data: PropTypes.string.isRequired
};

export default PlainText;
