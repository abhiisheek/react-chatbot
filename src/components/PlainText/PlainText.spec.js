import React from 'react';
import renderer from 'react-test-renderer';

import PlainText from './PlainText';

describe('PlainText', () => {
  it('Should have displayName set on the component', () => {
    expect(PlainText.displayName).toBeDefined();
  });

  it('Snapshot test', () => {
    const snapshot = renderer
      .create(<PlainText data="Hello message" />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
