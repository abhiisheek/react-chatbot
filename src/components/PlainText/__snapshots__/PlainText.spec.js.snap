// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainText Snapshot test 1`] = `
<div
  className="textMsg wrapper"
>
  Hello message
</div>
`;
